src/lib/utilities/request-from-api.ts

```diff
@@ -159,6 +182,27 @@ const withBearerToken = async (
   return headers;
 };
 
+const withIdToken = (
+  headers: HeadersInit,
+  idToken: string,
+  isBrowser = browser,
+): HeadersInit => {
+  // At this point in the code path, headers will always be set.
+  /* c8 ignore next */
+  if (!headers) headers = {};
+  if (!isBrowser) return headers;
```

Comment:
If this had `strict: true` in tsconfig this code could be made more obviously safe.


We can also default headers in the call with the `= {}` to skip the if (headers)  
```javascript
const withIdToken = (
	headers: HeadersInit = {},
	idToken: string,
	isBrowser = browser,
  )
```